lib/cli/ui/spinner.rb

```diff
@@ -9,11 +10,20 @@ module Spinner
       PERIOD = 0.1 # seconds
       TASK_FAILED = :task_failed
 
+      RUNES = %w(⠋ ⠙ ⠹ ⠸ ⠼ ⠴ ⠦ ⠧ ⠇ ⠏).freeze
+
       begin
-        runes = ['⠋', '⠙', '⠹', '⠸', '⠼', '⠴', '⠦', '⠧', '⠇', '⠏']
         colors = [CLI::UI::Color::CYAN.code] * 5 + [CLI::UI::Color::MAGENTA.code] * 5
-        raise unless runes.size == colors.size
-        GLYPHS = colors.zip(runes).map(&:join)
+        raise unless RUNES.size == colors.size
+        GLYPHS = colors.zip(RUNES).map(&:join)
+      end
+
+      class << self
+        attr_accessor(:index)
+
+        def current_rune
```

Comment:
I’m not a fan of introducing this global state. This can mess up SpinGroup or other multithreaded invocations.

Would there be a way to attach a Spinner instance to something like the Formatter?

Reply:
So we could do it that way, but the index is already set in `sg.wait`, not individually per spingroup item, and running two spingroups concurrently would be complete madness.

I went with this because it seemed simpler and I couldn't think of a case in practice that it would ever be an issue.

Reply:
Ok, that makes sense to me. Can you add a comment explaining that?